### ISSUE TITLE
The padding of max_pooling2d is not fixed as conv2d

### ISSUE THREAD

**Issue by RogerChern** [google-research/simclr#26]
https://github.com/google-research/simclr/blob/f3ca72f7efc085ad4abdb65f7a63459d9cfda78f/resnet.py#L487-L490

So there are two types of padding in this repo, which may cause confusion when users want to use simclr model in another codebase as the weights are neither compatible with the vanilla tensorflow nor other frameworks like pytorch or MXNet.

**Comment by chentingpc** [comment]
The resnet implementation in this repo should be consistent with other tensorflow official resnet implementation, such as [this](https://github.com/tensorflow/tpu/blob/master/models/official/resnet/resnet_model.py). Not too sure about MXNet, but the official pytorch [resnet](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py) also uses two types of padding and should also be compatible from what I can see.

**Comment by chentingpc** [comment]
btw, you may refer to this code base for conversion between the resnet here and pytorch: https://github.com/tonylins/simclr-converter

**Comment by RogerChern** [comment]
Thanks so much. I also converted the weights and found that the maxpooling in pytorch pad zero in all four borders while the maxpooling in tensorflow pad zero only the bottom and the right. 

Initially I think this slight difference makes the finetuning of simclr on pytorch failed. But after checking `https://github.com/tonylins/simclr-converter` I suddenly realized that simclr does not employ normalization. Now the checkpoint can be tuned normally.
